---
title: "Count body and end signal for the various samples and tracks"
author: "Manfred Schmid"
output: 
  pdf_document: 
    toc: true 
    toc_depth: 3
    fig_caption: true
---


## Setup

`r format(Sys.time(), "%d %B, %Y")`

```{r setup}
knitr::opts_chunk$set(echo = TRUE, fig.path = 'Figures_body_end_counting/', dev='pdf')
```


```{r, message=FALSE,warning=FALSE}
suppressWarnings(library('tidyverse'))
suppressWarnings(library('magrittr'))
suppressWarnings(library('knitr'))
suppressWarnings(library('broom'))
```


## LOAD 2min counts into R

```{r}
read_bedgraph_sums <- function(path, file_list, sfx) {
  lapply(file_list, function(fname) {read.table(paste0(path,fname,sep=''), header=F, col.names=c('chr', 'start', 'end', 'id', 'sum'), stringsAsFactors = FALSE) %>% 
      tbl_df %>%
  mutate(condition = sub(sfx, '', fname),
         condition = sub('norm_', '', condition),
         sum = ifelse(sum == '.', 0, as.numeric(sum)))}) %>%
  bind_rows()
}
```


#### load and combine Sp-normalized end counts
```{r}
path <- '~/Documents/Results/Lexogen_RNAseq_2/STAR_bw/norm_and_pA_filtered_bedgraph/bedtools_counts/'
sfx <- '_endpm200.*count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```


```{r, warning=FALSE}
(end_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### load and combine Sp-normalized body counts
```{r}
sfx <- '_body_endsm200.*_count_sum_per_anno.txt$'

file_list <- dir(path)[grep(sfx, dir(path))]

file_list
```

```{r, warning=FALSE}
(body_counts_df <- read_bedgraph_sums(path, file_list, sfx))
```


#### combine body and end 

```{r}
(both_df <- bind_rows(mutate(body_counts_df, part = 'body'), 
                     mutate(end_counts_df, part = 'end')) %>%
   mutate(pA = case_when(grepl('^noPap', .$condition) ~ 'pA+', 
                         grepl('^xPappAminus', .$condition) ~ 'pA-',
                         grepl('^xPappAmasked', .$condition) ~ 'pA-X',
                         grepl('^xPap_', .$condition) ~ 'pA+ + pA-',
                         TRUE ~ 'NA')))
```


```{r}
save(both_df, file='../../data/norm_2min_body_end_signals.RData')
```


## SANITY CHECK BGsub and pA minus method

simply checks that 
1. norm. counts for $BGsub = normIP  -  normnegativeIP$  
2. norm. counts for $pAminus = pAplusandminus  -  pAplus$

#### BGsub

```{r}
IP_raw <- filter(both_df, grepl('ip', condition) & !(grepl('BGsub', condition))) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop')

IP_neg <- filter(both_df, grepl('ip_neg0', condition)) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::select(-rapa, -rep) %>%
  dplyr::rename(ip_neg_sum = sum)

IP_BGsub <- filter(both_df, grepl('ip', condition) & (grepl('BGsub', condition))) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::rename(BGsub_sum = sum)
```


```{r}
IP_BGsub_redone <- left_join(IP_raw, IP_neg) %>%
  mutate(BGsub_sum_redone = sum - ip_neg_sum) %>%
  dplyr::select(-sum, -ip_neg_sum)
```

```{r}
(BGsub_comparison <- left_join(IP_BGsub_redone, IP_BGsub) %>%
   mutate(relative_error = (BGsub_sum_redone-BGsub_sum)/BGsub_sum))
```

```{r}
BGsub_comparison %>%
  filter(is.finite(BGsub_sum_redone), is.finite(BGsub_sum)) %>%
  group_by(pA, part, Pap, strain, rapa, rep) %>%
  #summarize(cnt=n()) %>% kable
  do(tidy(cor.test(.$BGsub_sum, .$BGsub_sum_redone))) %>%
  kable
```

```{r}
BGsub_comparison %>%
  filter(is.finite(BGsub_sum_redone), is.finite(BGsub_sum)) %>%
  group_by(pA, part, Pap, strain, rapa, rep) %>%
  do(tidy(summary(.$relative_error))) %>% 
  kable
```

--> perfectly correlated and same values
--> OK


#### pAminus

```{r}
pAplus <- filter(both_df, pA == 'pA+',
                 grepl('_ip_.*BGsub$', condition) | 
                   grepl('_input_', condition)) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::select(-Pap, -pA, -rep) %>%
  dplyr::rename(pAplus = sum)

pAplusandminus <- filter(both_df, pA == 'pA+ + pA-',
                 grepl('_ip_.*BGsub$', condition) | 
                   grepl('_input_', condition)) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::select(-Pap, -pA) %>%
  dplyr::rename(pAplusandminus = sum)

pAminus <- filter(both_df, pA == 'pA-',
                 grepl('_ip_.*BGsub$', condition) | 
                   grepl('_input_', condition)) %>%
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep'), sep='_', extra='drop') %>%
  dplyr::select(-Pap, -pA) %>%
  dplyr::rename(pAminus = sum)

```


```{r}
IP_pAminus_redone <- left_join(pAplus, pAplusandminus) %>%
  mutate(pAminus_redone = pAplusandminus - pAplus)
```

```{r}
(pA_comparison <- left_join(IP_pAminus_redone, pAminus) %>%
   mutate(relative_error = (pAminus-pAminus_redone)/pAminus_redone))
```

```{r}
pA_comparison %>%
  filter(is.finite(pAminus_redone), is.finite(pAminus)) %>%
  group_by(part, strain, fraction, rapa, rep) %>%
  #summarize(cnt=n()) %>% kable
  do(tidy(cor.test(.$pAminus_redone, .$pAminus))) %>%
  kable
```

```{r}
pA_comparison %>%
  filter(is.finite(pAminus_redone), is.finite(pAminus)) %>%
  group_by(part, strain, fraction, rapa, rep) %>%
  do(tidy(summary(.$relative_error))) %>% 
  kable
```

--> perfectly correlated and same values
--> OK



#### work only with BGsub IPs

```{r}
(both_df %<>% 
  separate(condition, c('Pap', 'strain', 'fraction', 'rapa', 'rep', 'BGsub'), sep='_') %>%
  separate(id, c('id', 'type', 'name', 'common_name'), sep=':') %>%
  filter(fraction == 'input' | (fraction == 'ip' & BGsub == 'BGsub'))  %>%
  dplyr::select(-BGsub))
```

Fix the XUT annotations, which are differently organized.
```{r}
both_df %<>%
  mutate(id = ifelse(id == 'XUT', type, id),
         type = ifelse(grepl('^XUT', type), 'XUT', type))
```


## TOTAL SIGNAL PER ANNOTATION TYPE

#### sum over body and end
```{r}
(total_sum <- both_df %>%
  dplyr::select(-chr, -start, -end) %>%
  spread(part, sum) %>%
  mutate(sum = body + end) %>%
  dplyr::select(-body, -end))
```

#### add SNR type and factor types and pA
```{r}
total_sum %<>%
  mutate(type = ifelse(grepl('^snR', name), 'SNR', type),
         type = factor(type, levels=c('ORF-T', 'SUTs', 'CUTs', 'SNR', 'other', 'XUT')),
         pA = factor(pA, levels=c('pA+', 'pA-', 'pA+ + pA-', 'pA-X')))
```


#### set negatives to 0
```{r}
total_sum %<>%
  mutate(sum = ifelse(sum < 0, 0, sum))
```


#### plot sums

```{r barplot sums per anno}
total_sum %>%
  ggplot(., aes(x=type, y=sum, fill=type)) +
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~strain+rapa+rep) +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw()
```


```{r barplot sums per anno clean rapa 0 Figure}
total_sum %>%
  filter(rapa == 0, 
         type != 'other', type != 'XUT', 
         pA == 'pA+' | pA == 'pA-') %>%
  ggplot(., aes(x=type, y=sum, fill=type)) +
  geom_bar(stat='identity') +
  facet_grid(pA+fraction~strain+rapa+rep) +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw()
```


#### mean over replicates and strains

```{r}
total_avg <- total_sum %>%
  group_by(pA, fraction, rapa, id, type) %>%
  summarize(avg = mean(sum)) %>%
  group_by(pA, fraction, rapa, type) %>%
  summarize(total = sum(na.omit(avg)))
```

```{r barplot averages per anno clean rapa 0 Figure}
total_avg %>%
  ungroup %>%
  filter(rapa == 0, 
         type != 'other', type != 'XUT', 
         pA == 'pA+' | pA == 'pA-') %>%
  ggplot(., aes(x=type, y=total, fill=type)) +
  geom_bar(stat='identity') +
  facet_grid(fraction~pA, scales='free') +
  scale_fill_brewer(palette = 'Set1') +
  theme_bw()
```


## NORMALIZE TO LENGTH

Normalize reads to region length and correct for genomic A masked positions..


### cleaned region lengths and A-masked positions

Note: for body regions stripped for overlapping ends, end regions are all 400bp. This info is contained in the A-mask count files.

A-masked positions: nothing will ever be counted from those positions, so in practice these positions do not count to the length of each region -> subtract them:  
```{r}
(effective_body_len <- read.table('~/Documents/Results/Lexogen_RNAseq/anno/clean_body_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'body_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(pAfiltered_cleaned_body_len = end - start - body_pA_masked))
```

```{r}
(effective_body_len_XUTs <- read.table('~/Documents/Results/Lexogen_RNAseq/anno/XUTs_cleaned2x_body_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'body_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('type', 'id'), sep=':') %>%
   mutate(pAfiltered_cleaned_body_len = end - start - body_pA_masked))
```

```{r}
(effective_end_len <- read.table('~/Documents/Results/Lexogen_RNAseq/anno/clean_ends_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'end_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':') %>%
   mutate(pAfiltered_end_len = end - start - end_pA_masked))
```

```{r}
(effective_end_len_XUTs <- read.table('~/Documents/Results/Lexogen_RNAseq/anno/XUTs_cleaned_ends_pAfilter.counts', col.names=c('chr', 'start', 'end', 'anno', 'end_pA_masked')) %>%
   tbl_df %>%
   separate(anno, c('type', 'id'), sep=':') %>%
   mutate(pAfiltered_end_len = end - start - end_pA_masked))
```


#### combine all of the above

```{r}
(effective_lens <- full_join(
  dplyr::select(effective_body_len, id, type, pAfiltered_cleaned_body_len),
  dplyr::select(effective_end_len, id, type, pAfiltered_end_len)) %>%
   arrange(id)
  )
```

```{r}
(effective_lens_XUTs <- full_join(
  dplyr::select(effective_body_len_XUTs, id, type, pAfiltered_cleaned_body_len),
  dplyr::select(effective_end_len_XUTs, id, type, pAfiltered_end_len)) %>%
   arrange(id)
  )
```




#### Final annotations, containing that info

Original plain annotations:  
```{r}
(anno <- read.table("~/Documents/genomewide_datasets/annotations/sacCer3/Steinmetz_sacCer3_latin.bed", col.names = c('chr', 'start', 'end', 'anno', 'pad', 'strand')) %>%
  tbl_df %>% 
   separate(anno, c('id', 'type', 'name', 'common_name'), sep=':')  %>%
   arrange(id))
```

 
```{r}
(anno <- bind_rows(effective_lens, effective_lens_XUTs) %>%
   full_join(anno, .) %>%
  mutate(type = ifelse(grepl('^snR', name), 'snRNA', as.character(type)),
         type = factor(type, levels=c('ORF-T', 'XUT', 'SUTs', 'CUTs', 'snRNA', 'other'))))
```


```{r}
(both_df %<>% 
  dplyr::select(id, sum, part, pA, Pap, strain, fraction, rapa, rep) %>%
  left_join(., anno) %>%
  mutate(scaled_reads = case_when(.$part == 'body' ~ sum/pAfiltered_cleaned_body_len,
                               .$part == 'end' ~ sum/pAfiltered_end_len)) %>%
  dplyr::select(id, name, common_name, type, part, pA, Pap, strain, fraction, rapa, rep, sum, scaled_reads))
```


#### save the normalized scaled signals

```{r}
save(both_df, file='../../data/norm_2min_body_end_scaled_and_BGsub.RData')
```




#### sanity check non-exisiting gene bodies
…. note some regions pAfiltered_cleaned_body_len are NA … ie there is no body, either completely overlapping with ends or otherwise the annotation is too short (or only consists of masked genomic A, although this probably never happens) !! 
-> there should be no reads for those
-> sanity check all with pAfiltered_cleaned_body_len == NA should never have counts

```{r}
body_len_NA_ids <- filter(anno, is.na(pAfiltered_cleaned_body_len)) %>% distinct(id, common_name, type)

table(body_len_NA_ids$type)
```

```{r}
(filter(effective_lens, is.na(pAfiltered_cleaned_body_len)) %$% type %>% table)

(filter(effective_lens_XUTs, is.na(pAfiltered_cleaned_body_len)) %>% nrow)
```

mostly CUTs, many SUTs and XUTs, some snRNAs very few ORFs.  


```{r}
filter(both_df, part == 'body', id %in% body_len_NA_ids$id)
```

```{r}
filter(both_df, part == 'end', id %in% body_len_NA_ids$id)
```


--> seems correct, we never count body for those where the region does not exist. But ends are apprently counted ...




#### look at some single genes

```{r}
filter(both_df,
       common_name == 'PMA1',
       rapa == 0, rep == 1) %>%
  kable
```

```{r}
filter(both_df,
       common_name == 'PGK1',
       rapa == 0, rep == 1) %>%
  kable
```


```{r}
filter(both_df,
       common_name == 'SNR11',
       rapa == 0, rep == 1) %>%
  kable
```

```{r}
filter(both_df,
       common_name == 'SCR1',
       rapa == 0, rep == 1) %>%
  kable
```



## AVERAGE VALUES


#### select only rapa 0 and pAplus or pAminus

```{r}
both_df %<>%
  filter(rapa == 0,
         pA == 'pA+' | pA == 'pA-')
```


## DR from ip relative input ratio

#### Step 1: average over replicates

```{r}
(avg_signal <- both_df %>%
  group_by(part, pA, strain, fraction, rapa, id, type, name) %>%
  summarize(mean_C = mean(scaled_reads)))
```


#### compare Mex vs Nab

```{r scatter plots Mex vs Nab average over replicates}
avg_signal %>%
  spread(strain, mean_C) %>%
  ggplot(., aes(x=Mex67AA, y=Nab2AA)) +
  geom_point() +
  scale_x_log10() + scale_y_log10() +
  facet_grid(part+pA~fraction+rapa)
```

overall correlated but there are differences especially in the IP samples...


#### Step 2: average strains
```{r}
(avg_signal  %<>%
  group_by(part, pA, fraction, rapa, id, type, name) %>%
  summarize(mean_C = mean(mean_C)))
```

#### save the average signals

```{r}
save(avg_signal, file='../../data/norm_2min_body_end_scaled_and_BGsub_avg_signals.RData')
```



#### plot average signal per type and sample

```{r violin plot avg signals per type part fraction and pA type rapa 0 pAplus and pAminus}
avg_signal %>%
  ungroup %>%
  mutate(part = factor(part, levels=c('body', 'end')),
         pA = factor(pA, levels=c('pA+', 'pA-')),
         fraction = factor(fraction, levels=c('input', 'ip')),
         type = factor(type, levels=c('ORF-T', 'SUTs', 'CUTs'))) %>%
  filter(!is.na(type), rapa == 0,
         pA == 'pA+' | pA == 'pA-') %>%
  ggplot(., aes(x=type, y=mean_C, fill=type)) +
  geom_violin() +
  geom_boxplot(width=.1, outlier.shape=NA, fill='lightgray') +
  facet_grid(part~pA+fraction+rapa) +
  scale_y_log10() +
  scale_fill_brewer(palette='Set1') +
  theme_bw()
```

n in the violins
```{r}
avg_signal %>%
  ungroup %>%
  mutate(part = factor(part, levels=c('body', 'end')),
         pA = factor(pA, levels=c('pA+', 'pA-')),
         fraction = factor(fraction, levels=c('input', 'ip')),
         type = factor(type, levels=c('ORF-T', 'SUTs', 'CUTs'))) %>%
  filter(!is.na(type), 
         rapa == 0,
         pA == 'pA+' | pA == 'pA-',
         mean_C > 0) %>%
  group_by(rapa, pA, part, fraction, type) %>%
  summarize(cnt = n()) %>%
  kable
```


```{r}
sessionInfo()
```



