---
title: "Quality Filtering  and Mapping"
author: "Manfred Schmid"
date: "9 August 2017"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Sequencing

Libraries were prepared using the QuantSeq REV kit from Lexogen (Lexogen GmbH, \link{https://www.lexogen.com}). Sequencing was done via Lexogen at the Biocenter Sequencing Facility Vienna using sequencing type: HiSeqV4 SR50.
Lexogen did barcode splitting and provided the final data as unmapped bam files (more compact than fastq, but contain same info).

Same procedure used for 2min 4tU and 10min 4tU experiments. Single examples depicted for demonstration are from the 2' 4tU series except if stated otherwise.



To inspect these bam files:

```{bash, eval = F}
samtools view 39035_GCAGCC_C9BADANXX_2_20160502B_20160502.bam | head
```

I got output of that kind
```{bash raw samtools view output, eval = F}
7001253F:369:C9BADANXX:2:2207:1337:2201#GCAGCC	4	*	0	0	*	*	0	0	TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTATTTTTTTTTTTTTTTTTTT	BBBBBFFFFFFFFFFFFFFFFFFFFFFFFF/<FFFFFB<//7//////7B	B2:Z:TCTTTCCCT	Q2:Z:BBBBBBBBB	BC:Z:GCAGCCATC	RG:Z:C9BADANXX.2#GCAGCC	QT:Z:BBBBBBBBB
7001253F:369:C9BADANXX:2:2207:1592:2083#GCAGCC	4	*	0	0	*	*	0	0	AATAATAGAAAAAGTTTTTTGAAAAAAAGAGAGCTGTTAAATATATAGGG	BBBBBF/<BFFFF<F<FFFFF<F///<7FF//<FFFFFF/FFFFFFFFF/	B2:Z:TCTTTCCCT	Q2:Z:BBBBBBBBB	BC:Z:GCAGCCATC	RG:Z:C9BADANXX.2#GCAGCC	QT:Z:BBBBBBBBB
7001253F:369:C9BADANXX:2:2207:1644:2109#GCAGCC	4	*	0	0	*	*	0	0	GTAGTAAGAACTATTCATAGAGTGAATCGATAACAATACGAAAATGTAAA	BBBBBFFFFFFFFFFFFFFFBFBFFFFFFF/<FF<FFFFFFFFFFFFFFF	B2:Z:TCTTTCCCT	Q2:Z:BBBBBBBBB	BC:Z:GCAGCCATC	RG:Z:C9BADANXX.2#GCAGCC	QT:Z:BBBBBBBBB
7001253F:369:C9BADANXX:2:2207:1570:2122#GCAGCC	4	*	0	0	*	*	0	0	AGTTTTATATACGTTGTATTAGAAATATAATATATGACAATAGTACACTT	BBBBBF<FBF<F<FFFFFFFFFBFFFFFFF/B<FFFFFFFFFFFFFFFFF	B2:Z:TCTTTCCCT	Q2:Z:BBBBBBBBB	BC:Z:GCAGCCATC	RG:Z:C9BADANXX.2#GCAGCC	QT:Z:BBBBBBBBB
7001253F:369:C9BADANXX:2:2207:2033:2142#GCAGCC	4	*	0	0	*	*	0	0	GAAACAATACGAAAATGTAAACATTTCCTAGACGTAGTATATAGAGACAA	BBBBBFFFFFFFFFFFFFBBBBFFFFFFFF<BFFFFFBFFFFFFBFFFFF	B2:Z:TCTTTCCCT	Q2:Z:BBBBBBBBB	BC:Z:GCAGCCATC	RG:Z:C9BADANXX.2#GCAGCC	QT:Z:BBBBBBBBB
7001253F:369:C9BADANXX:2:2207:2122:2213#GCAGCC	4	*	0	0	*	*	0	0
```



# QC and mapping
We used the QC filtering and mapping strategy and settings recommended by Lexogen at that time. The information was from (\link{https://www.lexogen.com/quantseq-data-analysis/}), however, note that I don't know whether the information from the link is stable and whether it has been changed since then.


Here is the code used:
```{bash, eval = FALSE}
#!/bin/sh
#call: sbatch --mem=16g -c 4 STAR_map.sh

##for file 42448_AATAGC_C9P6RANXX_5_20160808B_20160808
sample="42448_AATAGC_C9P6RANXX_5_20160808B_20160808"

cd /home/schmidm/faststorage/Lexogen2/bams

##convert raw to fastq 
source /com/extra/bedtools/2.16.2/load.sh

bamToFastq -i ${sample}.bam -fq ${sample}.fastq


##trim adapters (using bbduk.sh from BBmap suite, installed locally at ~/bin)
### from BBMap version 35.92
~/bin/bbmap/bbduk.sh in=${sample}.fastq out=${sample}_trimmed_clean.fastq ref=/home/schmidm/annotations/common/Lexogen_adapters_with_pA.fa.gz k=13 ktrim=r useshortkmers=t mink=5 qtrim=t trimq=10 minlength=20



##map using STAR

source /com/extra/samtools/1.3/load.sh
source /com/extra/STAR/git-2016-03-14/load.sh

STAR \
--runThreadN 4 \
--genomeDir /home/schmidm/annotations/sacCer3_Sp_merged_genome/STAR_index \
--readFilesIn ${sample}_trimmed_clean.fastq \
--outFilterType BySJout \
--outFilterMultimapNmax 20 \
--alignSJoverhangMin 8 \
--alignSJDBoverhangMin 1 \
--outFilterMismatchNmax 999 \
--alignIntronMin 20 \
--alignIntronMax 2000 \
--outFileNamePrefix ../STAR_map/${sample}_trimmed_clean \
--outSAMtype BAM SortedByCoordinate
```


#### The adapter filtering file use was as follows

It contains the usual Illumina adapter sequences + a polyA stretch as suggested by Lexogen. (The polyA is to remove quite abundant A-tails on reads when doing QuantSeq REV library protocol?)
```{bash, eval = FALSE}
gunzip /home/schmidm/annotations/common/Lexogen_adapters_with_pA.fa.gz
cat /home/schmidm/annotations/common/Lexogen_adapters_with_pA.fa

>TruSeq_Adapter_Index_1_6
GATCGGAAGAGCACACGTCTGAACTCCAGTCACATCACGATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_2
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCGATGTATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_3
GATCGGAAGAGCACACGTCTGAACTCCAGTCACTTAGGCATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_4
GATCGGAAGAGCACACGTCTGAACTCCAGTCACTGACCAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_5
GATCGGAAGAGCACACGTCTGAACTCCAGTCACACAGTGATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_6
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGCCAATATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_7
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCAGATCATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_8
GATCGGAAGAGCACACGTCTGAACTCCAGTCACACTTGAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_9
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGATCAGATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_10
GATCGGAAGAGCACACGTCTGAACTCCAGTCACTAGCTTATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_11
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGGCTACATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_12
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCTTGTAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_13
GATCGGAAGAGCACACGTCTGAACTCCAGTCACAGTCAACAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_14
GATCGGAAGAGCACACGTCTGAACTCCAGTCACAGTTCCGTATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_15
GATCGGAAGAGCACACGTCTGAACTCCAGTCACATGTCAGAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_16
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCCGTCCCGATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_18_7
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGTCCGCACATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_19
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGAAACGATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_20
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGTGGCCTTATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_21
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGTTTCGGAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_22
GATCGGAAGAGCACACGTCTGAACTCCAGTCACCGTACGTAATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_23
GATCGGAAGAGCACACGTCTGAACTCCAGTCACGAGTGGATATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_25
GATCGGAAGAGCACACGTCTGAACTCCAGTCACACTGATATATCTCGTATGCCGTCTTCTGCTTG
>TruSeq_Adapter_Index_27
GATCGGAAGAGCACACGTCTGAACTCCAGTCACATTCCTTTATCTCGTATGCCGTCTTCTGCTTG
>polyA
AAAAAAAAAAAAAAAAAA
```


#### genome used
Our samples are from S. cerevisiae and contain low amounts of total S.pombe RNA as spike-ins. I create an index file of a "merged" genome containing S. cerevisiae (UCSC release sacCer3) and S.pombe (ENSEMBL release EF2) for STAR mapping.
S. cerevisiae and S. pombe are easily separated since all S. cervisiae chromosomes are name "chr..." whereas the S. pombe do not have the "chr" prefix.

The index file was created as follows:
```{bash, eval = FALSE}
#!/bin/sh

source /com/extra/STAR/git-2016-03-14/load.sh

STAR \
--runThreadN 1 \
--runMode genomeGenerate \
--genomeDir /home/schmidm/annotations/sacCer3_Sp_merged_genome/STAR_index \
--genomeFastaFiles /home/schmidm/annotations/sacCer3_Sp_merged_genome/fasta/sacCer3_Sp_merged.fa \
--sjdbGTFfile /home/schmidm/annotations/sacCer3_Sp_merged_genome/anno/r64_2_1.gff3 \
--sjdbGTFtagExonParentTranscript Parent \
--sjdbGTFfeatureExon CDS \
--genomeSAindexNbases 11 \
--sjdbOverhang 49
```


#### indexing of the bam files
STAR output is sorted but not indexex. For indexing simply run:

```{bash, eval=F}
#!/bin/sh
cd /home/schmidm/faststorage/Lexogen/STAR_map

source /com/extra/samtools/1.3/load.sh

for f in *.bam; do samtools index $f; done
```



#### Collecting trimming and mapping stats:

Trimming stats can be derived from stdout (ie the *.out* file) from the job running bbduk like this:
its neither pretty nor perfect and leaves some fiddling to be done in Excel.
```{bash, eval=FALSE}
#!/bin/sh

cd /home/schmidm/faststorage/Lexogen2/STAR_map/scripts/job_out

#extract the jobs where bbmap succeeded (here has a QTrimmed row)
grep -rnl STAR_map_*.out -e "QTrimmed" > trim_and_map_jobs

#put the contents of these job.out into a single file ... not need in this version
##cat trim_and_map_jobs | while read f;do cat $f >> trim_and_map_job.outs ;done 

#cut down to the relevant rows
#if need be: rm tmp
cat trim_and_map_jobs | while read f;do echo $f >> tmp; cat $f | grep -e Input: -e QTrimmed -e KTrimmed -e Result | paste - - | paste - - >> tmp; done
 
cat tmp | paste - - > trim_and_map_job.outs_concise
rm tmp

#final processing in Excel ...
```



To get the mapping stats, kind of similar approach, but here the log file (*.final.out*) created by STAR contains the useful info. Again, its neither pretty nor perfect and leaves some fiddling to be done in Excel.
```{bash, eval=FALSE}
#!/bin/sh

cd /home/schmidm/faststorage/Lexogen/STAR_map

#put the final STAR map reports into a single file 
for f in *_trimmed_cleanLog.final.out; do echo $f >> STAR_final.outs; cat $f >> STAR_final.outs ;done 
#cut down to the relevant rows
cat STAR_final.outs | tr "\n" "\t" | sed s/"%\t390"/"%\n390"/g | cut -f1,11- > STAR_final.outs_concise
```

Final processing of the output from those scripts was done in Excel .



# Track files

The informative part of each read is the last position, ie the position before the polyA tail. I made a custom python script to map those into bedgraph track files. (The alternative version producing bigwigs directly is also available)

#### Lexogen_bam_to_bedgraph.py
```{python, eval = FALSE}
#!/usr/bin/env python
'''

usage:
      	python Lexogen_bam_to_bedgraph.py bam_file

For now just trying to get reads with NH tag = 1, obtain end position of each read, count them up and write directly to bedgraph file

'''


__author__ = 'schmidm'

import sys
import HTSeq
import numpy as np


if len(sys.argv) <= 1 or sys.argv[1] == '-h':
    print __doc__
    exit()

#open a bam file
bamfile = sys.argv[1]
bam_reader = HTSeq.BAM_Reader( bamfile )

plus_bg_fname = bamfile.replace('.bam', '_plus.bedgraph')
minus_bg_fname = bamfile.replace('.bam', '_minus.bedgraph')

#get header for chromosome information
chromosomes = {chunk['SN']: chunk['LN'] for chunk in bam_reader.get_header_dict()['SQ']}
print 'chromosomes from bam: ', chromosomes

#loop through the reads
i = 0
NH_included = 0
NH_excluded = 0
cur_chr = ''
plus_sum = 0
minus_sum = 0

def write_chr_values(bg_fname, chr, values):
    with open(bg_fname, 'a+') as f:
        start = 0
        end = 1
        value = 0
        for i,v in enumerate(values):            
            if i == end and v == value:
                #concatenate adjacent bins with identical value into larger interval
                end = i+1
            else:
                if value > 0:
                     f.write(chr + '\t' + str(start) + '\t' + str(end) + '\t' + str(value) + '\n')
                start = i
                end = i+1
                value = v
        if value > 0:
            f.write(chr + '\t' + str(start) + '\t' + str(end) + '\t' + str(value) + '\n')

for aln in bam_reader:
    if aln.optional_field('NH') != 1:
        NH_excluded += 1
        continue
    NH_included += 1
    iv = aln.iv

    if iv.chrom != cur_chr:
        #write out
        if cur_chr != '':
            write_chr_values(plus_bg_fname, cur_chr, plus_vals)
            write_chr_values(minus_bg_fname, cur_chr, minus_vals)
            plus_sum += sum(plus_vals)
            minus_sum += sum(minus_vals)
        #initialize next chr
        cur_chr = iv.chrom
        print 'processing reads for chr: ', cur_chr
        plus_vals = np.zeros(chromosomes[cur_chr], dtype = np.int)
        minus_vals = np.zeros(chromosomes[cur_chr], dtype = np.int)

    if iv.strand == '+':
        minus_vals[iv.start] += 1
    elif iv.strand == '-':
        plus_vals[iv.end - 1] += 1
    else:
        print 'missing strand info ...?'


write_chr_values(plus_bg_fname, cur_chr, plus_vals)
write_chr_values(minus_bg_fname, cur_chr, minus_vals)
plus_sum += sum(plus_vals)
minus_sum += sum(minus_vals)


## some output that helped in some initial debugging
print 'NH_excluded: ', NH_excluded
print 'NH_included: ', NH_included
print 'plus_sum: ', plus_sum
print 'minus_sum: ', minus_sum
```


Execution of the python script:  
```{bash, eval = FALSE}
#!/bin/sh
 
##NOTE: use local python with HTSeq installed

cd /home/schmidm/faststorage/Lexogen2/STAR_map
 
echo "doing bam_file $1"
python /home/schmidm/faststorage/Lexogen/STAR_map/scripts/Lexogen_bam_to_bedgraph.py $1
echo "done"

##job
#cd /home/schmidm/faststorage/Lexogen2/STAR_map
#for f in *.bam; do sbatch --mem=2g -c 1 scripts/bam_to_bedgraph.sh $f; done
```


HTSeq version used:  
```{bash, eval = FALSE}
python
Python 2.7.11 |Continuum Analytics, Inc.| (default, Dec  6 2015, 18:08:32) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
>>> import HTSeq
>>> print HTSeq.__version__
0.6.0
>>> exit()
```


# Change name to informative names

The bam and bedgraph file names are not informative. I changed filenames using a table with this little bash script.
```{bash, eval = FALSE}
#!usr/bin/bash

map=$1
in_sfx=$2
out_sfx=$3

cat $map | while read barcode_line; do barcode_map=(${barcode_line/ / }); \
mv ${barcode_map[1]}${in_sfx} ${barcode_map[0]}${out_sfx}; done
```


For the 10 minutes 4tU experiment data the table used is:
```{bash, eval = FALSE}
cat file_rename_map_correct.txt
Nab_in_0_1	39007_ACATTA_C9BADANXX_1_20160502B_20160502
Nab_in_15_1	39008_GAACCT_C9BADANXX_1_20160502B_20160502
Nab_in_70_1	39009_ACAACG_C9BADANXX_1_20160502B_20160502
Nab_in_0_2	39010_AGTTGA_C9BADANXX_1_20160502B_20160502
Nab_in_15_2	39011_AGGCAT_C9BADANXX_1_20160502B_20160502
Nab_in_70_2	39012_GAAGTG_C9BADANXX_1_20160502B_20160502
Nab_in_0_3	39013_AACAAG_C9BADANXX_1_20160502B_20160502
Nab_in_15_3	39014_TTGGTA_C9BADANXX_1_20160502B_20160502
Nab_in_70_3	39015_GGTGAG_C9BADANXX_1_20160502B_20160502
Nab_in_neg	39016_CGGTTA_C9BADANXX_1_20160502B_20160502
Mex_in_0_1	39017_GCGCTG_C9BADANXX_1_20160502B_20160502
Mex_in_15_1	39018_GACGAT_C9BADANXX_1_20160502B_20160502
Mex_in_70_1	39019_ACCTAC_C9BADANXX_1_20160502B_20160502
Mex_in_0_2	39020_AGAATC_C9BADANXX_1_20160502B_20160502
Mex_in_15_2	39021_AACCGA_C9BADANXX_1_20160502B_20160502
Mex_in_70_2	39022_CAACAG_C9BADANXX_1_20160502B_20160502
Mex_in_0_3	39023_CGAAGG_C9BADANXX_1_20160502B_20160502
Mex_in_15_3	39024_AACGCC_C9BADANXX_1_20160502B_20160502
Mex_in_70_3	39025_CAAGCA_C9BADANXX_1_20160502B_20160502
Mex_in_neg	39026_CACACT_C9BADANXX_1_20160502B_20160502
Nab_ip_0_1	39027_TGGATT_C9BADANXX_1_20160502B_20160502
Nab_ip_15_1	39028_GCGAAT_C9BADANXX_1_20160502B_20160502
Nab_ip_70_1	39029_TGGCGA_C9BADANXX_1_20160502B_20160502
Nab_ip_0_2	39030_CAATGC_C9BADANXX_1_20160502B_20160502
Nab_ip_15_2	39031_AAGACA_C9BADANXX_2_20160502B_20160502
Nab_ip_70_2	39032_CAGATG_C9BADANXX_2_20160502B_20160502
Nab_ip_0_3	39033_GTTACC_C9BADANXX_2_20160502B_20160502
Nab_ip_15_3	39034_CAGCGT_C9BADANXX_2_20160502B_20160502
Nab_ip_70_3	39035_GCAGCC_C9BADANXX_2_20160502B_20160502
Nab_ip_neg	39036_CGATCT_C9BADANXX_2_20160502B_20160502
Mex_ip_0_1	39037_CACTAA_C9BADANXX_2_20160502B_20160502
Mex_ip_15_1	39038_GGAGGT_C9BADANXX_2_20160502B_20160502
Mex_ip_70_1	39039_TAATCG_C9BADANXX_2_20160502B_20160502
Mex_ip_0_2	39040_GATCAC_C9BADANXX_2_20160502B_20160502
Mex_ip_15_2	39041_CTCTCG_C9BADANXX_2_20160502B_20160502
Mex_ip_70_2	39042_TGCTAT_C9BADANXX_2_20160502B_20160502
Mex_ip_0_3	39043_CGCCTG_C9BADANXX_2_20160502B_20160502
Mex_ip_15_3	39044_CATCTA_C9BADANXX_2_20160502B_20160502
Mex_ip_70_3	39045_AAGCTC_C9BADANXX_2_20160502B_20160502
Mex_ip_neg	39046_CAGGAC_C9BADANXX_2_20160502B_20160502
noPap_in_0	39047_CGCAAC_C9BADANXX_2_20160502B_20160502
noPap_in_70	39048_CGCGGA_C9BADANXX_2_20160502B_20160502
noPap_ip_0	39049_CCAATT_C9BADANXX_2_20160502B_20160502
noPap_ip_70	39050_TCTTAA_C9BADANXX_2_20160502B_20160502
xPap_in_0	39051_CCGACC_C9BADANXX_2_20160502B_20160502
xPap_in_70	39052_AAGTGG_C9BADANXX_2_20160502B_20160502
xPap_ip_0	39053_TACCTT_C9BADANXX_2_20160502B_20160502
xPap_ip_70	39054_GGCCAA_C9BADANXX_2_20160502B_20160502
```


For the 2 minutes 4tU experiment data the table used is:
```{bash, eval = FALSE}
cat ../../file_rename_map_Lexogen2.txt
noPap_Nab2AA_input_0_1 42448_AATAGC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Nab2AA_input_15_1 42449_CCTAAG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_input_0_1 42450_TTCGAG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_input_15_1 42451_CCGCAA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_input_70_1 42452_TATGTC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Nab2AA_ip_0_1 42453_TGCACG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Nab2AA_ip_15_1 42454_CTAGTC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_ip_neg0_1 42455_CTCATA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_ip_0_1 42456_TTAACT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_ip_15_1 42457_GGCTGC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
noPap_Mex67AA_ip_70_1 42458_CGTCGC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_input_0_1 42459_CTCCAT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_input_15_1 42460_TGACAC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_input_0_1 42461_TCGTTC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_input_15_1 42462_AATCCG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_input_70_1 42463_CCTGCT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_0_1 42464_AATGAA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_0_2 42465_ACCAGT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_0_3 42466_TGTGCA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_15_1 42467_GTCAGG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_15_2 42468_ACAGAT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Nab2AA_ip_15_3 42469_ACACGC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_neg0_1 42470_GTGTAG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_0_1 42471_GGTATA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_0_2 42472_GATTGT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_0_3 42473_GTGCCA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_15_1 42474_ACCGTG_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_15_2 42475_ACGTCT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_15_3 42476_AGACCA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_70_1 42477_GTAGAA_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_70_2 42478_ACTCTT_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
xPap_Mex67AA_ip_70_3 42479_TTCCGC_C9P6RANXX_5_20160808B_20160808_trimmed_cleanAligned.sortedByCoord.out
```

run the renaming script:  

10min data
```{bash, eval = FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq/STAR_bw/raw_bedgraph

bash ../../scripts/barcode_file_renamer.sh ../../file_rename_map_correct.txt _trimmed_cleanAligned.sortedByCoord.out_plus.bedgraph _plus.bedgraph

bash ../../scripts/barcode_file_renamer.sh ../../file_rename_map_correct.txt _trimmed_cleanAligned.sortedByCoord.out_minus.bedgraph _minus.bedgraph
```

2min data
```{bash, eval = FALSE}
cd /Users/schmidm/Documents/Results/Lexogen_RNAseq_2/STAR_bw/raw_bedgraph

bash ../../scripts/barcode_file_renamer.sh ../../file_rename_map_Lexogen2.txt _plus.bedgraph _plus.bedgraph

bash ../../scripts/barcode_file_renamer.sh ../../file_rename_map_Lexogen2.txt _minus.bedgraph _minus.bedgraph
```